lib/active_merchant/billing/gateways/nuvei.rb

```diff
@@ -150,6 +154,64 @@ def add_address(post, payment, options)
         }.compact
       end
 
+      def execute_3ds_flow(post, money, payment, transaction_type, options = {})
+        post_3ds = post.dup
+
+        MultiResponse.run do |r|
+          r.process { commit(:init_payment, post) }
+
+          r.process do
+            three_d_params = r.params.dig('paymentOption', 'card', 'threeD')
+            three_d_supported = three_d_params['v2supported'] != 'false'
+
+            next r.process { Response.new(false, '3D Secure is required but not supported') } if !three_d_supported && options[:force_3d_secure]
+
+            if three_d_supported
+              add_3ds_data(post_3ds, options.merge(version: three_d_params['version']))
+              post_3ds[:relatedTransactionId] = r.authorization
+            end
+
+            commit(:purchase, post_3ds)
+          end
+        end
+      end
+
+      def add_3ds_data(post, options = {})
+        three_d_secure = options[:three_ds_2]
+        # 01 => Challenge requested, 02 => Exemption requested, 03 or not sending parameter => No preference
+        challenge_preference = if options[:force_3d_secure] == true || options[:force_3d_secure] == 'true'
+                                 '01'
+                               elsif options[:force_3d_secure] == false || options[:force_3d_secure] == 'false'
```

Comment:
Small suggestion, you could make this more idiomatic to Ruby by using `include?` like so:

`if [true, 'true'].include?(options[:force_3d_secure]`